File: app/src/main/java/com/botijasoftware/utils/FontText.java
package com.botijasoftware.utils;

public class FontText {

    private int textLen;
    private String oldText;
    private Font mFont;
    private static final int MAX_CHARS = 32;
    private SpriteBatch spritebatch;
    private Sprite tmpSprite;
    private float fontsize;
    private float fontsizesep;



	public FontText(Font font)
	{
	    mFont = font;
		textLen = 0;


	}
	
	public void LoadContent(ResourceManager resources) {
	
		float x = 0.0f;
		float y = 0.0f;
		fontsize = (float) mFont.getFontSize();// * mLayout.getHorizontal(1);// * fontsize * 0.2f;
        fontsizesep = fontsize * 3/7;

        spritebatch = new SpriteBatch(MAX_CHARS);
        tmpSprite = new Sprite(new Rectanglef(0,0,fontsize,fontsize), mFont.getTexture());
	}

    public void Draw(float x, float y, String text) {
        Draw(x, y, text, null);
    }

	
	public void Draw(float x, float y, String text, ColorRGBAb color) {

		spritebatch.begin();


        textLen = text.length();

        spritebatch.begin();

        if (textLen > MAX_CHARS)
            textLen = MAX_CHARS;

        if (color != null)
            tmpSprite.setColor(color);

        int ascii = 0;

        for (int i = 0; i< textLen; i++) {

            ascii = text.codePointAt(i);
            tmpSprite.setTexture(mFont.getTexture(ascii));
            tmpSprite.getRectangle().setPosition(x+ i*fontsizesep , y);
            spritebatch.DrawSprite(tmpSprite);

        }
        spritebatch.end();

	}

    public Vector2 getTextSize(String text) {
        return new Vector2(fontsizesep*text.length(), fontsize);
    }


}

File: app/src/main/java/com/botijasoftware/utils/FontText.kt
package com.botijasoftware.utils

class FontText(private val mFont: Font) {

    private var textLen: Int = 0
    private lateinit var oldText: String
    private lateinit var spritebatch: SpriteBatch
    private lateinit var tmpSprite: Sprite
    private var fontsize: Float = 0.toFloat()
    private var fontsizesep: Float = 0.toFloat()


    init {
        textLen = 0


    }

    fun LoadContent(resources: ResourceManager) {

        val x = 0.0f
        val y = 0.0f
        fontsize = mFont.fontSize.toFloat()// * mLayout.getHorizontal(1);// * fontsize * 0.2f;
        fontsizesep = fontsize * 3 / 7

        spritebatch = SpriteBatch(MAX_CHARS)
        tmpSprite = Sprite(Rectanglef(0f, 0f, fontsize, fontsize), mFont.texture)
    }


    @JvmOverloads fun Draw(x: Float, y: Float, text: String, color: ColorRGBAb? = null) {

        spritebatch.begin()


        textLen = text.length

        spritebatch.begin()

        if (textLen > MAX_CHARS)
            textLen = MAX_CHARS

        if (color != null)
            tmpSprite.setColor(color)

        var ascii: Int

        for (i in 0..textLen - 1) {

            ascii = text.codePointAt(i)
            tmpSprite.texture = mFont.getTexture(ascii)
            tmpSprite.rectangle.setPosition(x + i * fontsizesep, y)
            spritebatch.DrawSprite(tmpSprite)

        }
        spritebatch.end()

    }

    fun getTextSize(text: String): Vector2 {
        return Vector2(fontsizesep * text.length, fontsize)
    }

    companion object {
        private val MAX_CHARS = 32
    }


}
